lib/fileHandler/index.js

```diff
@@ -324,42 +335,68 @@ module.exports = function (options) {
 		};
 	}
 
+	function checkExpiryConditions (url, headers, rawBody) {
```

Comment:
This doesn't seem like the ideal way to check if a preSignedUrl has expired. 
I think we should be looking at the query params of the URL. 
So for Signature Version 2 it has an `Expires` and for signature v4 `X-Amz-Date`

Also I don't think it's a good idea to look for the `amazonaws` domain, we could switch to an s3 alternative in future or use a custom domain for our s3 bucket.

Reply:
This seems like a good to identify if the URL has expired. Guess we'd need to provision for URLs that don't include these params (in case people provide a file object with a URL that doesn't point to S3). In that case we would just reject with the usual error.